File: core/src/main/java/org/bnb/dw/core/Question.java
package org.bnb.dw.core;

import java.util.Set;

public class Question {

  public final Noun noun;
  public final Set<Choice> choices;

  public Question(Noun noun, Set<Choice> choices) {
    this.noun = noun;
    this.choices = choices;
  }

  public String text() {
    return noun.word;
  }
}

File: core/src/main/kotlin/org/bnb/dw/core/Question.kt
package org.bnb.dw.core

class Question(val prototype: QuestionPrototype, val choices: List<Choice>) {

    fun text(): String {
        return when (prototype.type) {
            QuestionType.GENDER -> prototype.noun.word
            QuestionType.TERM_ITSELF -> prototype.noun.translation
            QuestionType.TRANSLATION -> prototype.noun.gender.article + " " + prototype.noun.word
        }
    }
}
